watson_developer_cloud/watson_service.py

```diff
@@ -292,6 +290,9 @@ def set_username_and_password(self, username=None, password=None):
     def set_api_key(self, api_key):
         if api_key == 'YOUR API KEY':
             api_key = None
+        if api_key is not None and api_key.startswith(ICP_PREFIX):
+            self.set_username_and_password(APIKEY, api_key)
+            return
```

Comment:
@germanattanasio Should we add this to the other SDKs too? I didn't think about this but if users try to authenticate using an icp apikey the SDK does basicauth instead?